test/unit/lib/handlers/quote/handler.test.ts

```diff
@@ -227,6 +229,79 @@ describe('QuoteHandler', () => {
         expect(quoteJSON.quoteGasAdjusted).toBe(CLASSIC_QUOTE_EXACT_IN_WORSE.amountOutGasAdjusted.toString());
       });
 
+      it('check request source', async () => {
+        const quoteMock = jest.fn().mockResolvedValue(CLASSIC_QUOTE_EXACT_IN_WORSE)
+        const quoterMock: Quoter = { quote: quoteMock }
+
+        const quoters = { [RoutingType.CLASSIC]: quoterMock };
+        const tokenFetcher = TokenFetcherMock([TOKEN_IN, TOKEN_OUT]);
+        const portionFetcher = PortionFetcherMock(GET_NO_PORTION_RESPONSE);
+        const permit2Fetcher = Permit2FetcherMock(PERMIT_DETAILS);
+        const syntheticStatusProvider = SyntheticStatusProviderMock(false);
+
+        const quoteHandler = getQuoteHandler(
+          quoters,
+          tokenFetcher,
+          portionFetcher,
+          permit2Fetcher,
+          syntheticStatusProvider
+        )
+
+        let headers: APIGatewayProxyEventHeaders = {
+          'x-request-source': 'uniswap-web',
+        }
+        await quoteHandler.handler(getEvent(CLASSIC_REQUEST_BODY, headers), {} as unknown as Context);
+        let quoteCallParams = quoteMock.mock.lastCall[0]
+        expect(quoteCallParams.info.source).toBe(RequestSource.UNISWAP_WEB)
+      });
+
+      it('test getQuoteRequestSource', async () => {
+        const quoteMock = jest.fn().mockResolvedValue(CLASSIC_QUOTE_EXACT_IN_WORSE)
+        const quoterMock: Quoter = { quote: quoteMock }
+
+        const quoters = { [RoutingType.CLASSIC]: quoterMock };
+        const tokenFetcher = TokenFetcherMock([TOKEN_IN, TOKEN_OUT]);
+        const portionFetcher = PortionFetcherMock(GET_NO_PORTION_RESPONSE);
+        const permit2Fetcher = Permit2FetcherMock(PERMIT_DETAILS);
+        const syntheticStatusProvider = SyntheticStatusProviderMock(false);
+
+        const quoteHandler = getQuoteHandler(
+          quoters,
+          tokenFetcher,
+          portionFetcher,
+          permit2Fetcher,
+          syntheticStatusProvider
+        )
+
+        let headers: APIGatewayProxyEventHeaders = { 'x-request-source': 'uniswap-ios' }
+        let requestSource = quoteHandler.getQuoteRequestSource(headers)
+        expect(requestSource).toBe(RequestSource.UNISWAP_IOS)
```

Comment:
If we do end up lowercasing the headers during comparison we should add a test here to test the case insensitivity